### PR TITLE
mattupham/ad banner

### PR DESCRIPTION
**Pull request by mattupham**
## What is the purpose of the change

- we want to promote other projects, add an ad banner above the trade clipboard

### ClickUp Task

[ClickUp Task URL](https://app.clickup.com/t/86785q0zc)

## Brief Changelog

- add a mock CMS
- add a rotating ad banner
- add dynamic styles for image, text color, arrow color, and background gradient
- added LD flag "swaps-ad-banner"

## Testing and Verifying

https://github.com/osmosis-labs/osmosis-frontend/assets/30577966/858c328c-fc9e-4474-8122-0f48e16407e8

https://github.com/osmosis-labs/osmosis-frontend/assets/30577966/0bca22da-df48-48d2-aa6b-b0403b3cc72b

### CODE REVIEW

File: packages/web/public/icons/arrow-right-icon.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    />
+    <path
+      d="M14 6L19 12L14 18"
+      stroke={stroke}
+      strokeWidth="2.5"
+      strokeLinecap="round"
+      strokeLinejoin="round"
+    />
+  </svg>
+);
```
[the remark this answers is not in the document]
Comment (jonator) [reply packages/web/public/icons/arrow-right-icon.tsx:24]: Icon.tsx wouldn't work here?

**Review by jonator — COMMENTED** [review]
Initial review since build was failing.

Nit: for CL I added a components/funnels folder that I think, categorically, this could all fit nicely into. Having good folder names, and less of them, makes intuitive code navigation easier for new devs.

**Review by jonator — COMMENTED** [review]
It seems there needs to be some custom changes to the stepper dots, they're overflowing above the component:
Screenshot 2023-06-26 at 7 47 57 PM

File: packages/web/components/ad-banner/ad-banner.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+export const AdBanner = memo(() => {
+  const featuredAds = useMemo(
+    () => adCMS.banners.filter(({ featured }) => featured),
+    []
+  );
+  const ads = useMemo(() => shuffleArray(featuredAds), [featuredAds]);
+
+  return (
+    <Stepper autoplay={{ delayInMs: 12000, stopOnHover: true }}>
```
Comment (MaxMillington) [inline packages/web/components/ad-banner/ad-banner.tsx:21]: Did we confirm this is the right delay that product wants?

File: packages/web/pages/index.tsx [hunk 3]
```diff
@@ -10,6 +11,13 @@ import { useStore } from "~/stores";
 import { EventName, IS_FRONTIER, IS_TESTNET } from "../config";
 import { useAmplitudeAnalytics } from "../hooks";
 
+const AdBanner = dynamic<{}>(
+  () => import("../components/ad-banner").then((module) => module.AdBanner),
+  {
+    ssr: false,
+  }
+);
+
```
[the remark this answers is not in the document]
Comment (MaxMillington) [reply packages/web/pages/index.tsx:21]: Interesting. Maybe you could share this during the dev meeting?

**Review by MaxMillington — APPROVED** [review]

**Review by jonator — APPROVED** [review]
Approved, but we still need to figure out where JSON is stored. I think @MaxMillington and I agree it should be on GH via Airtable, or some similar solution that allows no coders to update it on the fly.

**Pull request merged**